src/hardwaresigner/mod.rs

```diff
@@ -0,0 +1,69 @@
+//! Hardware wallets
+//!
+//! This module contains modules required for working with Hardware Wallets
+
+pub mod signer;
+
+#[cfg(test)]
+mod tests {
+    use std::{str::FromStr, sync::Arc};
+
+    use bitcoin::{Address, Network};
+    use hwi::{types, HWIClient};
+
+    use crate::{
+        database::MemoryDatabase, hardwaresigner::signer::HWISigner, signer::SignerOrdering,
+        KeychainKind, SignOptions, Wallet,
+    };
+
+    fn get_first_device() -> HWIClient {
+        HWIClient::get_client(
+            HWIClient::enumerate().unwrap().first().expect(
+                "No devices found. Either plug in a hardware wallet, or start a simulator.",
+            ),
+            true,
+            types::HWIChain::Test,
+        )
+        .unwrap()
+    }
+
+    #[test]
+    fn test_create_signer() {
```

Comment:
In BDK we have two different types of tests:
- unit tests
- integration tests, which use blockchain backends

This should probably be a unit test, and should have the same structure as the other ones. See, for example, `test_sign_single_xprv` in `src/wallet/mod.rs`: there's no blockchain backend being created, only the transaction creation/signing part is tested. This test should also probably be moved in `src/wallet/mod.rs`.

Then, integration tests could be useful as well. To write one, you should modify the `bdk_blockchain_tests` in `src/testutils/blockchain_tests.rs`. Those tests will be run against every backend we support.